From 2411a3e2b5dfa4784bbe2548f8ef0ce1aea0379e Mon Sep 17 00:00:00 2001
From: Jason Jin <Jason.jin@freescale.com>
Date: Thu, 4 Aug 2011 09:59:55 +0800
Subject: [PATCH 50/52] Update the DMA map function for CF platform

In the DMA map, only the dcache need to be flushed.
Flush both dcache and icache will decrease the system
performance.

Signed-off-by: Jason Jin <Jason.jin@freescale.com>
---
 arch/m68k/kernel/dma.c |   12 ++++++++++--
 1 files changed, 10 insertions(+), 2 deletions(-)

--- a/arch/m68k/kernel/dma.c
+++ b/arch/m68k/kernel/dma.c
@@ -94,10 +94,18 @@ void dma_sync_single_for_device(struct d
 {
 	switch (dir) {
 	case DMA_TO_DEVICE:
-		cache_push(handle, size);
+#if defined(CONFIG_M5445X) || defined(CONFIG_M5441X)
+		flush_dcache_range(handle, handle + size);
+#else
+		flush_dcache();
+#endif
 		break;
 	case DMA_FROM_DEVICE:
-		cache_clear(handle, size);
+#if defined(CONFIG_M5445X) || defined(CONFIG_M5441X)
+		cf_dcache_inv_range(handle, handle + size);
+#else
+		flush_dcache();
+#endif
 		break;
 	case PCI_DMA_BIDIRECTIONAL:
 		flush_dcache();
